controllers/metal3.io/host_state_machine.go

```diff
@@ -227,6 +235,37 @@ func (hsm *hostStateMachine) checkInitiateDelete() bool {
 	return true
 }
 
+func (hsm *hostStateMachine) checkDetachedHost(info *reconcileInfo) (result actionResult) {
+	// If the detached annotation is set we remove any host from the
+	// provisioner and take no further action
+	// Note this doesn't change the current state, only the OperationalStatus
+	annotations := hsm.Host.GetAnnotations()
+	if annotations != nil {
+		if _, ok := annotations[metal3v1alpha1.DetachedAnnotation]; ok {
+			// Only allow detaching hosts in Provisioned/ExternallyProvisioned states
+			if info.host.Status.Provisioning.State == metal3v1alpha1.StateProvisioned || info.host.Status.Provisioning.State == metal3v1alpha1.StateExternallyProvisioned {
```

Comment:
nit: I think a switch statement would be much clearer here

Reply:
Yeah, done!